pkg/integration/tests/cherry_pick/cherry_pick_on_branch.go

```diff
@@ -0,0 +1,99 @@
+package cherry_pick
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var CherryPickOnBranch = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Paste cherry-picked commits onto branch from branches view",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.
+			EmptyCommit("base").
+			NewBranch("first-branch").
+			NewBranch("second-branch").
+			Checkout("first-branch").
+			EmptyCommit("one").
+			EmptyCommit("two").
+			Checkout("second-branch").
+			EmptyCommit("three").
+			EmptyCommit("four")
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		t.Views().Branches().
+			Focus().
+			Lines(
+				Contains("second-branch"),
+				Contains("first-branch"),
+				Contains("master"),
+			).
+			PressEnter()
+
+		t.Views().SubCommits().
+			IsFocused().
+			Lines(
+				Contains("four").IsSelected(),
+				Contains("three"),
+				Contains("base"),
+			).
+			// copy commits 'four' and 'three'
+			Press(keys.Commits.CherryPickCopy).
+			Tap(func() {
+				t.Views().Information().Content(Contains("1 commit copied"))
+			}).
+			SelectNextItem().
+			Press(keys.Commits.CherryPickCopy)
+
+		t.Views().Information().Content(Contains("2 commits copied"))
+
+		t.Views().Branches().
+			Focus().
+			Lines(
+				Contains("second-branch").IsSelected(),
+				Contains("first-branch"),
+				Contains("master"),
+			).
+			SelectNextItem().
+			Press(keys.Commits.PasteCommits).
+			Tap(func() {
+				// cherry-picked commits will be deleted after confirmation
```

Comment:
What does it mean that they will be _deleted_ after confirmation?

Is that referring to the selection, that it will be cleared or am I missing something?